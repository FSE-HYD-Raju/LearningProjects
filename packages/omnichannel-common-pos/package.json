{
	"name": "omnichannel-common-pos",
	"version": "111.2.0",
	"description": "Common actions, stores and components for omnichannel",
	"main": "lib/index",
	"types": "lib/index.d.ts",
	"oc:es6": "src/index",
	"scripts": {
		"test": "../../testSinglePackage.sh $(pwd)",
		"test-watch": "jest --watch",
		"lint": "eslint tests src --max-warnings=0 ",
		"test-ci": "jest",
		"lint-fix": "eslint src tests --fix",
		"flow": "flow; test $? -eq 0 -o $? -eq 2",
		"clean": "rm -rf lib",
		"build-ts": "tsc --pretty",
		"copy-assets": "copyfiles --up 1 'src/**/*.@(scss|jpg|jpeg|gif|png|svg)' ./lib",
		"build": "cross-env NODE_ENV=production BABEL_ENV=production BABEL_DISABLE_CACHE=1 babel -s inline -d lib/ src/ && npm run build-ts && npm run copy-assets && rimraf -rf messages",
		"watch": "cross-env NODE_ENV=production BABEL_ENV=production BABEL_DISABLE_CACHE=1 babel -w -s inline -d lib/ src/ && rimraf -rf messages",
		"prepublish": "npm run clean && npm run build"
	},
	"jest": {
		"testRegex": "(/tests/.*)\\.test\\.js$",
		"coverageDirectory": "jestcoverage",
		"coverageReporters": [
			"lcov",
			"html"
		],
		"setupFiles": [
			"../../jest-setup.js"
		]
	},
	"author": "",
	"license": "ISC",
	"publishConfig": {
		"registry": "https://artifactory.qvantel.net/artifactory/api/npm/npm-local"
	},
	"dependencies": {
		"alt": "0.18.6",
		"alt-utils": "2.0.0",
		"autosuggest-highlight": "2.1.1",
		"axios": "0.18.0",
		"babel-cli": "6.26.0",
		"babel-core": "6.26.0",
		"babel-eslint": "7.2.3",
		"babel-jest": "23.6.0",
		"babel-loader": "7.1.3",
		"babel-plugin-add-module-exports": "0.2.1",
		"babel-plugin-module-resolver": "3.1.0",
		"babel-plugin-react-intl": "2.4.0",
		"babel-plugin-react-transform": "2.0.2",
		"babel-plugin-transform-class-properties": "6.24.1",
		"babel-plugin-transform-decorators-legacy": "1.3.4",
		"babel-plugin-transform-es2015-classes": "6.24.1",
		"babel-plugin-transform-proto-to-assign": "6.26.0",
		"babel-plugin-transform-react-constant-elements": "6.23.0",
		"babel-plugin-transform-react-inline-elements": "6.22.0",
		"babel-plugin-transform-react-pure-class-to-function": "1.0.1",
		"babel-plugin-transform-react-remove-prop-types": "0.4.20",
		"babel-plugin-transform-runtime": "6.23.0",
		"babel-plugin-typecheck": "3.9.0",
		"babel-polyfill": "6.26.0",
		"babel-preset-es2015": "6.24.1",
		"babel-preset-es2015-loose": "7.0.0",
		"babel-preset-react": "6.24.1",
		"babel-preset-stage-0": "6.24.1",
		"babel-register": "6.26.0",
		"babel-runtime": "6.26.0",
		"bootstrap-sass": "3.3.7",
		"byte-size": "4.0.4",
		"classnames": "2.2.6",
		"connected-react-router": "4.3.0",
		"copyfiles": "2.0.0",
		"country-data": "0.0.31",
		"cross-env": "5.1.4",
		"crypto-js": "3.1.9-1",
		"css-ns": "1.2.1",
		"debug": "4.1.0",
		"diff": "3.4.0",
		"history": "4.7.2",
		"jwt-decode": "2.2.0",
		"langs": "2.0.0",
		"lodash": "4.17.11",
		"moment": "2.21.0",
		"moment-duration-format": "2.2.2",
		"object-assign": "4.1.1",
		"omnichannel-flow-pos": "111.2.0",
		"path": "0.12.7",
		"path-to-regexp": "2.3.0",
		"prop-types": "15.6.1",
		"ramda": "0.24.1",
		"rc-progress": "2.2.5",
		"rc-slider": "8.6.1",
		"react": "16.10.2",
		"react-autosuggest": "9.3.4",
		"react-bootstrap": "0.32.1",
		"react-collapse": "4.0.3",
		"react-confirm": "0.1.14",
		"react-dates": "20.1.0",
		"react-dom": "16.10.2",
		"react-dropzone": "3.13.4",
		"react-flatpickr": "3.6.4",
		"react-formal": "0.25.4",
		"react-formal-inputs": "2.1.1",
		"react-helmet": "5.2.0",
		"react-highlight-words": "0.11.0",
		"react-input-range": "1.3.0",
		"react-intl": "2.7.0",
		"react-motion": "0.5.2",
		"react-motion-ui-pack": "0.10.3",
		"react-onclickoutside": "6.7.1",
		"react-redux": "5.0.7",
		"react-router": "4.2.0",
		"react-router-dom": "4.2.2",
		"react-select": "1.2.1",
		"react-toastr": "3.0.0",
		"react-widgets": "3.5.0",
		"recompose": "0.30.0",
		"redux": "4.0.0",
		"redux-saga": "0.16.0",
		"redux-saga-tester": "1.0.379",
		"shallowequal": "0.2.2",
		"twix": "1.2.1",
		"universal-cookie": "2.1.2",
		"uuid": "3.2.1",
		"weasel-styles": "2.3.1",
		"yup": "0.21.3"
	},
	"devDependencies": {
		"@types/alt": "0.16.35",
		"@types/classnames": "2.2.6",
		"@types/jest": "23.3.9",
		"@types/lodash": "4.14.118",
		"@types/moment-duration-format": "2.2.2",
		"@types/node": "8.0.0",
		"@types/rc-slider": "8.2.3",
		"@types/react": "16.7.18",
		"@types/react-bootstrap": "0.32.14",
		"@types/react-collapse": "4.0.2",
		"@types/react-confirm": "0.1.2",
		"@types/react-dates": "17.1.5",
		"@types/react-dom": "16.0.5",
		"@types/react-flatpickr": "3.7.0",
		"@types/react-helmet": "5.0.7",
		"@types/react-highlight-words": "0.12.0",
		"@types/react-intl": "2.3.9",
		"@types/react-motion": "0.0.27",
		"@types/react-motion-ui-pack": "0.10.1",
		"@types/react-onclickoutside": "6.0.4",
		"@types/react-redux": "6.0.0",
		"@types/react-router": "4.0.25",
		"@types/react-select": "1.2.0",
		"@types/react-toastr": "3.0.3",
		"@types/recompose": "0.26.4",
		"@types/redux-actions": "2.3.0",
		"@types/redux-devtools": "3.0.44",
		"@types/redux-devtools-dock-monitor": "1.1.32",
		"@types/redux-devtools-log-monitor": "1.0.33",
		"@types/redux-logger": "3.0.6",
		"@types/seamless-immutable": "7.1.2",
		"@types/universal-cookie": "2.2.0",
		"@types/webpack": "4.4.0",
		"@types/webpack-env": "1.13.6",
		"@types/yup": "0.26.0",
		"babel-plugin-__coverage__": "11.0.0",
		"babel-plugin-transform-flow-strip-types": "6.22.0",
		"babel-plugin-transform-runtime": "6.23.0",
		"enzyme": "3.7.0",
		"enzyme-adapter-react-16": "1.10.0",
		"enzyme-to-json": "3.3.3",
		"eslint": "4.10.0",
		"eslint-config-omnichannel": "2.2.4",
		"eslint-config-prettier": "1.7.0",
		"eslint-config-react-app": "2.1.0",
		"eslint-loader": "1.9.0",
		"eslint-plugin-babel": "4.1.2",
		"eslint-plugin-flowtype": "2.39.1",
		"eslint-plugin-import": "2.8.0",
		"eslint-plugin-jsx-a11y": "5.1.1",
		"eslint-plugin-prettier": "2.6.0",
		"eslint-plugin-react": "7.4.0",
		"eslint-plugin-react-intl": "1.1.1",
		"flow-bin": "0.70.0",
		"jest": "23.6.0",
		"jest-junit": "3.6.0",
		"lint-staged": "4.3.0",
		"moxios": "0.4.0",
		"pre-commit": "1.2.2",
		"prettier": "1.6.1",
		"react-router-enzyme-context": "1.2.0",
		"react-test-renderer": "16.7.0",
		"redux-devtools": "3.4.1",
		"redux-devtools-dock-monitor": "1.1.3",
		"redux-devtools-log-monitor": "1.4.0",
		"redux-logger": "3.0.6",
		"rimraf": "2.6.2",
		"sinon": "4.3.0",
		"ts-jest": "23.10.5",
		"ts-loader": "5.3.1",
		"typescript": "3.1.6",
		"webpack": "4.26.1"
	},
	"commitInfo": "omni-ui-common-pos|184f176|1571059113|Venkata_Gangadhar_Kodela|develop"
}
