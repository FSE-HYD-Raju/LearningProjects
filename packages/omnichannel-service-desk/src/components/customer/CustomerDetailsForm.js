import PropTypes from "prop-types";
import Form from "react-formal";
import {
	withFormal,
	cssns,
	Flex,
	contextTypesValidationMap,
	OcDatePicker,
	LanguagesUtil,
	FormattedMessage,
	OcSelect,
	OcTextInput, OcButton, OcButtonType,
} from "omnichannel-common-pos";
import messages from "../../index.messages";
import customerMessages from "./Customer.messages";
import moment from "moment";
import { DELETE_DATA_OPTION } from "./CustomerDetailsMainView";

const { React } = cssns("CustomerDetailsView");
const momentLocalizer = require("react-widgets/lib/localizers/moment");
momentLocalizer(moment);

const FormalInput = withFormal(OcTextInput);
const FormalSelect = withFormal(OcSelect);
const uniqueLanguageCodes = LanguagesUtil.uniqueLanguageCodes;


const getInputType = field => {
	if (field.type === "select") {
		return FormalSelect;
	}
	return FormalInput;
};

const CustomerDetailsForm = (props, context) => {
	const handleSubmit = model => {
		props.modelSchema
			.validate(model)
			.then(validModel => {
				console.log("VALID MODEL::", validModel);
				try {
					props.onSubmit(validModel);
				} catch (e) {
					/* without this try-catch an error in props.onSubmit() is catched in the .catch() below,
					 * which would be disinformational.
					 */
					console.error(e);
				}
			})
			.catch(err => {
				console.log("INVALID MODEL::", err);
			});
	};

	const { formatMessage } = context.intl;

	const renderOption = (option, idx) => {
		return (
			<option
				key={`option_${idx}`}
				value={option.code || option}
				id={`CustomerDetailsDemographicInfoEdit-option-${option.code}`}
			>
				{option.name || option}
			</option>
		);
	};

	const renderHiddenOption = (option, idx) => {
		return (<option
			key={`option_${idx}`}
			value="not_defined"
			id="CustomerDetailsDemographicInfoEdit-option-not-defined"
			hidden={true}
		>
			{ formatMessage(messages.notDefined) }
		</option>);
	};

	const renderOptions = field => {
		if (field.fieldName === "language") {
			return uniqueLanguageCodes(field.options).map((option, idx) => {
				if((option.code || option) === DELETE_DATA_OPTION) {
					return renderHiddenOption(option, idx);
				} else {
					return renderOption(option, idx);
				}
			});
		} else {
			return (
				field.options &&
				field.options.map((option, idx) => {
					return renderOption(option, idx);
				})
			);
		}
	};
	return (
		<div>
			<Form
				schema={props.modelSchema}
				defaultValue={props.model}
				onSubmit={handleSubmit}
			>
				{props.fields.map((field, idx) => {
					const inputType = getInputType(field);

					return (
						<Flex
							key={`${field.fieldName}-${idx}`}
							style={{ marginBottom: "2px", height: "50px" }}
							alignItems="center"
						>
							{field.type === "date" ? (
								<Form.Field
									id={`CustomerDetailsForm-datepicker-${field.fieldName}`}
									required={field.required}
									name={field.fieldName}
									max={new Date()}
									placeholder={
										!field.value ? (
											formatMessage(messages.chooseADate)
										) : (
											moment(field.value).format("LL")
										)
									}
									label={formatMessage({
										...messages[field.fieldName]
									})}
									labelPosition="left"
									labelWidth="124px"
									type={OcDatePicker}
									format="LL"
									editFormat="LL"
									withClock={false}
									dropUp={false}
									events={["onSubmit"]}
								/>
							) : (
								<Form.Field
									id={
										field.type === "select" ? (
											`CustomerDetailsForm-select-${field.fieldName}`
										) : (
											`CustomerDetailsForm-input-${field.fieldName}`
										)
									}
									required={field.required}
									name={field.fieldName}
									label={
										<FormattedMessage
											{...messages[field.fieldName]}
										/>
									}
									labelPosition="left"
									labelWidth="124px"
									type={inputType}
									events={["onBlur", "onChange", "onSubmit"]}
								>
									{renderOptions(field)}
								</Form.Field>
							)}
						</Flex>
					);
				})}

				<Flex
					justifyContent="end"
					alignItems="center"
					className="action-buttons"
				>
					{props.updatingUser && (
						<i
							className="fa fa-spin fa-circle-o-notch"
							style={{
								fontSize: "16px",
								marginRight: "8px",
								color: "#aaa"
							}}
						/>
					)}

					<Form.Button type="submit" component="div">
						<OcButton
							id={`CustomerDetailsForm-submit-${props.name}`}
							htmlBtnType="submit"
							buttonType={OcButtonType.SUCCESS}
							disabled={props.updatingUser}
						>
							<FormattedMessage {...customerMessages.formSaveButton} />
						</OcButton>
					</Form.Button>
				</Flex>
			</Form>
		</div>
	);
};

CustomerDetailsForm.propTypes = {
	modelSchema: PropTypes.object,
	model: PropTypes.object,
	onSubmit: PropTypes.func,
	onChange: PropTypes.func,
	onCancel: PropTypes.func,
	fields: PropTypes.array,
	name: PropTypes.string,
	updatingUser: PropTypes.bool
};

CustomerDetailsForm.contextTypes = contextTypesValidationMap;

export default CustomerDetailsForm;
