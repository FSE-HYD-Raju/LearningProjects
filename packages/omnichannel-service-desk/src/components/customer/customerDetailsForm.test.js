import React from "react";
import yup from "yup";
import { mountWithContext, shallowWithContext } from "omnichannel-common-pos";

import CustomerDetailsForm from "../../../src/components/customer/CustomerDetailsForm";

describe("CustomerDetailsForm", () => {
	it("succeeds at shallow mount with minimal props", () => {
		const wrapper = shallowWithContext(<CustomerDetailsForm fields={[]} />);
		expect(wrapper).toMatchSnapshot();
	});

	it("succeeds at deep mount with minimal props", () => {
		mountWithContext(<CustomerDetailsForm fields={[]} />);
	});

	describe("RUBT-65825", () => {
		const fields = [
			{
				fieldName: "street",
				value: "1 test st",
				required: true
			},
			{
				fieldName: "postalCode",
				value: "12340",
				required: false
			},
			{
				fieldName: "city",
				value: "Testia",
				required: true
			},
			{
				fieldName: "country",
				value: "Testland",
				type: "select",
				options: [],
				required: true
			}
		];

		const props = {
			fields,
			modelSchema: yup.object({
				street: yup.string().required(),
				postalCode: yup.string(),
				city: yup.string().required(),
				country: yup.string().required()
			})
		};

		it("renders an asterisk representing field mandatoriness", () => {
			const wrapper = mountWithContext(
				<CustomerDetailsForm {...props} />
			);

			const fieldElements = wrapper.find("Field");
			expect(
				fieldElements.filterWhere(
					n => n.find(".fa-asterisk").length > 0
				).length
			).toEqual(fields.filter(f => f.required).length);
		});

		it("renders an asterisk representing field mandatoriness -- snapshot", () => {
			const wrapper = shallowWithContext(
				<CustomerDetailsForm {...props} />
			);
			expect(wrapper).toMatchSnapshot();
		});
	});
});
