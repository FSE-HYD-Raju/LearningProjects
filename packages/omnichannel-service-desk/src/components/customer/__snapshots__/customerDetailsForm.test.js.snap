// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomerDetailsForm RUBT-65825 renders an asterisk representing field mandatoriness -- snapshot 1`] = `
<div>
  <Uncontrolled(Form)
    onSubmit={[Function]}
    schema={
      ObjectSchema {
        "_blacklist": BadSet {
          "_map": Object {},
          "_refs": Object {},
        },
        "_conditions": Array [],
        "_defaultDefault": [Function],
        "_deps": Array [],
        "_excludedEdges": Array [
          "undefined-undefined",
        ],
        "_exclusive": Object {},
        "_mutate": false,
        "_nodes": Array [
          "country",
          "city",
          "postalCode",
          "street",
        ],
        "_options": Object {
          "abortEarly": true,
          "recursive": true,
        },
        "_type": "object",
        "_typeError": [Function],
        "_whitelist": BadSet {
          "_map": Object {},
          "_refs": Object {},
        },
        "fields": Object {
          "city": StringSchema {
            "_blacklist": BadSet {
              "_map": Object {},
              "_refs": Object {},
            },
            "_conditions": Array [],
            "_deps": Array [],
            "_exclusive": Object {
              "required": false,
            },
            "_mutate": false,
            "_options": Object {
              "abortEarly": true,
              "recursive": true,
            },
            "_type": "string",
            "_typeError": [Function],
            "_whitelist": BadSet {
              "_map": Object {},
              "_refs": Object {},
            },
            "tests": Array [
              [Function],
              [Function],
            ],
            "transforms": Array [
              [Function],
            ],
          },
          "country": StringSchema {
            "_blacklist": BadSet {
              "_map": Object {},
              "_refs": Object {},
            },
            "_conditions": Array [],
            "_deps": Array [],
            "_exclusive": Object {
              "required": false,
            },
            "_mutate": false,
            "_options": Object {
              "abortEarly": true,
              "recursive": true,
            },
            "_type": "string",
            "_typeError": [Function],
            "_whitelist": BadSet {
              "_map": Object {},
              "_refs": Object {},
            },
            "tests": Array [
              [Function],
              [Function],
            ],
            "transforms": Array [
              [Function],
            ],
          },
          "postalCode": StringSchema {
            "_blacklist": BadSet {
              "_map": Object {},
              "_refs": Object {},
            },
            "_conditions": Array [],
            "_deps": Array [],
            "_exclusive": Object {},
            "_mutate": false,
            "_options": Object {
              "abortEarly": true,
              "recursive": true,
            },
            "_type": "string",
            "_typeError": [Function],
            "_whitelist": BadSet {
              "_map": Object {},
              "_refs": Object {},
            },
            "tests": Array [],
            "transforms": Array [
              [Function],
            ],
          },
          "street": StringSchema {
            "_blacklist": BadSet {
              "_map": Object {},
              "_refs": Object {},
            },
            "_conditions": Array [],
            "_deps": Array [],
            "_exclusive": Object {
              "required": false,
            },
            "_mutate": false,
            "_options": Object {
              "abortEarly": true,
              "recursive": true,
            },
            "_type": "string",
            "_typeError": [Function],
            "_whitelist": BadSet {
              "_map": Object {},
              "_refs": Object {},
            },
            "tests": Array [
              [Function],
              [Function],
            ],
            "transforms": Array [
              [Function],
            ],
          },
        },
        "tests": Array [],
        "transforms": Array [
          [Function],
        ],
      }
    }
  >
    <Flex
      alignItems="center"
      alignSelf=""
      justifyContent=""
      key="street-0"
      style={
        Object {
          "height": "50px",
          "marginBottom": "2px",
        }
      }
      wrap="wrap"
    >
      <Field
        events={
          Array [
            "onBlur",
            "onChange",
            "onSubmit",
          ]
        }
        exclusive={false}
        id="CustomerDetailsForm-input-street"
        label={
          <FormattedMessageWrapper
            defaultMessage="Street address"
            description="service-desk, new customer, label street address"
            id="newcustomer-street-address"
            wrapperType="wrapper"
          />
        }
        labelPosition="left"
        labelWidth="124px"
        name="street"
        required={true}
        type={[Function]}
      />
    </Flex>
    <Flex
      alignItems="center"
      alignSelf=""
      justifyContent=""
      key="postalCode-1"
      style={
        Object {
          "height": "50px",
          "marginBottom": "2px",
        }
      }
      wrap="wrap"
    >
      <Field
        events={
          Array [
            "onBlur",
            "onChange",
            "onSubmit",
          ]
        }
        exclusive={false}
        id="CustomerDetailsForm-input-postalCode"
        label={
          <FormattedMessageWrapper
            defaultMessage="Postal code"
            description="service-desk, new customer, label postal code"
            id="newcustomer-psotal-code"
            wrapperType="wrapper"
          />
        }
        labelPosition="left"
        labelWidth="124px"
        name="postalCode"
        required={false}
        type={[Function]}
      />
    </Flex>
    <Flex
      alignItems="center"
      alignSelf=""
      justifyContent=""
      key="city-2"
      style={
        Object {
          "height": "50px",
          "marginBottom": "2px",
        }
      }
      wrap="wrap"
    >
      <Field
        events={
          Array [
            "onBlur",
            "onChange",
            "onSubmit",
          ]
        }
        exclusive={false}
        id="CustomerDetailsForm-input-city"
        label={
          <FormattedMessageWrapper
            defaultMessage="City"
            description="service-desk, new customer, label city"
            id="newcustomer-city"
            wrapperType="wrapper"
          />
        }
        labelPosition="left"
        labelWidth="124px"
        name="city"
        required={true}
        type={[Function]}
      />
    </Flex>
    <Flex
      alignItems="center"
      alignSelf=""
      justifyContent=""
      key="country-3"
      style={
        Object {
          "height": "50px",
          "marginBottom": "2px",
        }
      }
      wrap="wrap"
    >
      <Field
        events={
          Array [
            "onBlur",
            "onChange",
            "onSubmit",
          ]
        }
        exclusive={false}
        id="CustomerDetailsForm-select-country"
        label={
          <FormattedMessageWrapper
            defaultMessage="Country"
            description="service-desk, new customer, label country"
            id="newcustomer-country"
            wrapperType="wrapper"
          />
        }
        labelPosition="left"
        labelWidth="124px"
        name="country"
        required={true}
        type={[Function]}
      />
    </Flex>
    <Flex
      alignItems="center"
      alignSelf=""
      className="CustomerDetailsView-action-buttons"
      justifyContent="end"
      wrap="wrap"
    >
      <Button
        component="div"
        events={
          Array [
            "onClick",
          ]
        }
        type="submit"
      >
        <OcButton
          buttonType="SUCCESS"
          htmlBtnType="submit"
          id="CustomerDetailsForm-submit-undefined"
        >
          <FormattedMessageWrapper
            defaultMessage="Save"
            description="Save button for customer details form"
            id="customer-details-form-save-button"
            wrapperType="wrapper"
          />
        </OcButton>
      </Button>
    </Flex>
  </Uncontrolled(Form)>
</div>
`;

exports[`CustomerDetailsForm succeeds at shallow mount with minimal props 1`] = `
<div>
  <Uncontrolled(Form)
    onSubmit={[Function]}
  >
    <Flex
      alignItems="center"
      alignSelf=""
      className="CustomerDetailsView-action-buttons"
      justifyContent="end"
      wrap="wrap"
    >
      <Button
        component="div"
        events={
          Array [
            "onClick",
          ]
        }
        type="submit"
      >
        <OcButton
          buttonType="SUCCESS"
          htmlBtnType="submit"
          id="CustomerDetailsForm-submit-undefined"
        >
          <FormattedMessageWrapper
            defaultMessage="Save"
            description="Save button for customer details form"
            id="customer-details-form-save-button"
            wrapperType="wrapper"
          />
        </OcButton>
      </Button>
    </Flex>
  </Uncontrolled(Form)>
</div>
`;
